### PR TITLE
JITM: Support JITM message CTA links

### PR DESCRIPTION
**Pull request by kallehauge**
### Description

JITM messages can (through the JITM API) contain CTA specific links that doesn't use Jetpack Redirects. It's typically not recommended to bypass Jetpack Redirects but it can be needed if we wish to link directly to internal pages (a specific case could be a JITM that redirects a user to the Jetpack Partner Coupon redeem screen).

### Changes proposed in this Pull Request:

* Support CTA links as well as JITM message URLs (read: allow links outside Jetpack Redirects).

### Jetpack product discussion

* PT: pbtFPC-2c3-p2
* Prerequisite for specific task: 1201265858818217-as-1201439117106187 (D71432-code)

### Does this pull request change what data or activity we track or use?

* No.

### Testing instructions:

* Follow the test instructions over at D71470-code (contains a lot of internal lingo).

### CODE REVIEW

**Review by jboland88 — COMMENTED** [review]
Looks like a nice simple change - just added one question.

File: projects/packages/jitm/src/js/jetpack-jitm.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ jQuery( document ).ready( function ( $ ) {
 				html += '<div class="jitm-banner__action">';
 				html +=
 					'<a href="' +
-					envelope.url +
+					( envelope.CTA.link.length ? envelope.CTA.link : envelope.url ) +
```
[the remark this answers is not in the document]
Comment (kallehauge) [reply projects/packages/jitm/src/js/jetpack-jitm.js]: AFAIK yes. All properties for CTA should always exist but just with empty/false values. I have - however - added a `envelope.CTA.hasOwnProperty( 'link' )` condition just to make the condition more defensive 😄 

FYI: what cannot be seen in the code diff is that the rest of the JS code hard-assumes that `envelope.CTA` exists. E.g.: `if ( envelope.CTA.message )` is the whole condition determining if a CTA should be rendered or not, so the whole thing should have failed before we reach this code, if the CTA object doesn't exist 😃

**Comment by kallehauge** [comment]
Moving it from "Needs Team Review" to "Needs Review" since @jboland88 approved it in Asana.
Josh, feel free to leave an approved review here as well 😄

**Comment by sdixon194** [comment]
Hmm, tried testing this one, but can't seem to get it to work. I can get the JITM to appear, but never getting it pointed to `google.com`

- Made sure to sandbox my domain, wordpress.com and jetpack.com
- pulled down D71470-code 
- on a JN site with this branch applied, set JETPACK__SANDBOX_DOMAIN to my sandbox domain.
- made sure to enable global-http
- Still shows the same URL: 

Screen Shot on 2021-12-20 at 14:53:55

I'm sure I'm just missing something in testing, but I'm not sure what 🤔

**Comment by kallehauge** [comment]
@sdixon194 👋  I've tried to walk through the instructions to ensure nothing has broken since I created it but it seems to work (you know... for me 😂 )

I've updated the WPCOM diff to also change the title of the JITM to `Sandboxed - Get the new Jetpack mobile app.`. Would you mind testing it again? The title should work as an indicator that you're getting the modified version of the JITM before trying to debug further 😛

**Comment by sdixon194** [comment]
> I've updated the WPCOM diff to also change the title of the JITM to Sandboxed - Get the new Jetpack mobile app.. Would you mind testing it again? The title should work as an indicator that you're getting the modified version of the JITM before trying to debug further 😛

Hmm, strange, I don't see the updated JITM at all. I wonder if I'm not sandboxing something correctly? jetpack.com, my test site, wordpress.com, public-api.wordpress.com - that it?

**Review by sdixon194 — APPROVED** [review]

**Pull request merged**